contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -4235,6 +4239,88 @@ pg_proc_update_oid_acl(ObjectAddress address, Oid oid, Acl *acl)
 	table_close(rel, RowExclusiveLock);
 }
 
+/*
+ * Update the function definition of an alter procedure query 
+ * "ALTER ..." to "CREATE ..." in bbf_function_ext
+ */
+static void bbf_func_ext_update_proc_definition(Oid oid)
+{
+	Relation	bbf_function_ext_rel;
+	TupleDesc	bbf_function_ext_rel_dsc;
+	Datum 		new_record[BBF_FUNCTION_EXT_NUM_COLS];
+	bool		new_record_replaces[BBF_FUNCTION_EXT_NUM_COLS];
+	bool		new_record_nulls[BBF_FUNCTION_EXT_NUM_COLS];
+	char 		*original_query = get_original_query_string();
+	HeapTuple	tuple,
+				proctup,
+				oldtup;
+	
+	
+	proctup = SearchSysCache1(PROCOID, ObjectIdGetDatum(oid));
+	if (!HeapTupleIsValid(proctup)) // throw error
+		return;
+	oldtup = get_bbf_function_tuple_from_proctuple(proctup);
+	bbf_function_ext_rel = table_open(get_bbf_function_ext_oid(), RowExclusiveLock);
+	bbf_function_ext_rel_dsc = RelationGetDescr(bbf_function_ext_rel);
+
+	MemSet(new_record_nulls, false, sizeof(new_record_nulls));
+	MemSet(new_record_replaces, false, sizeof(new_record_replaces));
+
+	if (HeapTupleIsValid(oldtup))
+	{
+		StringInfoData infoSchemaStr;
+
+		initStringInfo(&infoSchemaStr);
+
+		for(int i = 0; i < strlen(original_query); i++)
+		{
+			if(original_query[i] == '-')
+			{
+				while(original_query[i] != '\n' && i < strlen(original_query))
+				{
+					appendStringInfoChar(&infoSchemaStr, original_query[i]);
+					i++;
+				}
+				appendStringInfoChar(&infoSchemaStr, original_query[i]);
+			}
+			else if(original_query[i] == '/' && original_query[i + 1] == '*' && i < strlen(original_query))
+			{
+				while(original_query[i] != '*' && original_query[i+1] != '/')
+				{
+					appendStringInfoChar(&infoSchemaStr, original_query[i]);
+					i++;
+				}
+				appendStringInfoChar(&infoSchemaStr, original_query[i]);
+				appendStringInfoChar(&infoSchemaStr, original_query[i+1]);
+				i++;
+			}
+			else if(strncasecmp(original_query + i, "alter", 5) == 0)
```

Comment:
strncasecmp didn't guarantee it'll check the end of string, before the if , should also make sure it didn't reach the boundary, like i+5 < strlen()